lib/smart_name.rb

```diff
@@ -107,6 +217,31 @@ def decoded
     @decoded ||= (s.index('&') ?  HTMLEntities.new.decode(s) : s)
   end
 
+  def pretty_key
+    parts.map do |part|
+      part.to_name.key.split('_').map do |kp|
+        if kp =~ /^[A-Z]/ && num_string = Base64.decode64($'+ '=')
```

Comment:
Oddly, this like throws a really strange error without padding with '='.  The encoder pads with '=' to the next multiple of 4 characters, and I didn't want to store any of those characters in the keys.  Later ruby versions decode just fine without the =, so this patch can probably go away at some point.